shared/api/auth.go

```diff
@@ -24,3 +24,140 @@ const (
 	// IdentityTypeOIDCClient represents an identity that authenticates with OIDC..
 	IdentityTypeOIDCClient = "OIDC client"
 )
+
+// Identity is the type for an authenticated party that can make requests to the HTTPS API.
+//
+// swagger:model
+type Identity struct {
+	// AuthenticationMethod is the authentication method that the identity
+	// authenticates to LXD with.
+	// Example: tls
+	AuthenticationMethod string `json:"authentication_method" yaml:"authentication_method"`
+
+	// Type is the type of identity.
+	// Example: oidc-service-account
+	Type string `json:"type" yaml:"type"`
+
+	// Identifier is a unique identifier for the identity (e.g. certificate fingerprint or email for OIDC).
+	// Example: jane.doe@example.com
+	Identifier string `json:"id" yaml:"id"`
+
+	// Name is the Name claim of the identity if authenticated via OIDC, or the name
+	// of the certificate if authenticated with TLS.
+	// Example: Jane Doe
+	Name string `json:"name" yaml:"name"`
+}
+
+// IdentityInfo expands an Identity to include group membership.
+//
+// swagger:model
+type IdentityInfo struct {
+	IdentityPut `yaml:",inline"`
+	Identity    `yaml:",inline"`
+}
+
+// IdentityPut contains the editable fields of an IdentityInfo.
+//
+// swagger:model
+type IdentityPut struct {
```

Comment:
Why cant you update `Name` by this struct?

Reply:
Because the `Name` of an OIDC identity is retrieved from the identity tokens `name` claim (requested via the `profile` scope) and should not be changed. The name of a certificate can be changed via the certificates API.

Reply:
cool thanks